airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -135,7 +156,7 @@ def execute(self, context: Context) -> None:
                 raise e
             finally:
                 if err is None:
-                    _upload_file_to_s3(f, self.s3_bucket_name, self.s3_key_prefix, self.aws_conn_id)
+                    _upload_file_to_s3(f, self.s3_bucket_name, self.s3_key_prefix, str(self.dest_aws_conn_id))
```

Comment:
Why we need string representation of `self.dest_aws_conn_id` here?

Reply:
Ohhh, You could change `_upload_file_to_s3` signature, no problem here at all this is private function (so you could do whatever you wanted and required), which for some reason initially in Airflow 1.x implemented outside of the Operator, and signature `str | None` expected for S3Hook.

Reply:
@Taragolis if I were to change the signature of `_upload_file_to_s3` to extend the type of `aws_conn_id` to `str | ArgNotSet | None` to match `dest_aws_conn_id`, I have a problem with S3Hook which only expects `str | None` and we're passing `dest_aws_conn_id` there.

I tried this inside constructor, but that didn't help me:
```python
        if dest_aws_conn_id is NOTSET:
            self.dest_aws_conn_id = self.source_aws_conn_id
        elif isinstance(dest_aws_conn_id, str):
            self.dest_aws_conn_id = dest_aws_conn_id
        else:
            self.dest_aws_conn_id = None
```

Strangely enough, ArgNotSet part of the type still escapes. Would you have any suggestions?

Reply:
TBH, at this point I don't understand why do you suggest `NOTSET` sentinel as a default fallback for new connections instead of using the default fallback of `aws_default` like it is widely done throughout this provider? 😅 

It would still support passing over the `None` value to `boto3`, right?

Reply:
@Taragolis I wanted to make it clear that I'm blocked here 😅

Reply:
`None` is a valid value for `aws_conn_id` so you can't use for resolve between deprecated and current value `None`, you shoul use NOTSET, or other sentinels.

And main thing you need to resolve before pass any values to upstream.

Reply:
I guess this is for avoid `mypy` complaining in static checks?

Reply:
Correct, to be able to remove type casting, which wasn't correct (it would produce string from a None value).

Reply:
Hey @Taragolis, thank you for your suggestions, I happily included them. Sorry for not noticing test failures.